### PR TITLE
Adding add_bigoh methods to lazy power/Laurent series and coercions to the finite precision implementations

### PR DESCRIPTION
**Pull request by tscrim**
It is natural to go from the lazy series (infinite precision) to the finite precision cases. We provide these coercion maps and the analogous methods `add_bigoh` and `O` for better compatibility.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

**Comment by mantepse** [comment]
There is a doctest failure in `laurent_series_ring.py`

**Comment by tscrim** [comment]
Fixed.

File: src/sage/rings/lazy_series.py [hunk 1]
```diff
@@ -6065,6 +6068,44 @@ def polynomial(self, degree=None, names=None):
             return R(self[0:m])
         return R.sum(self[0:m])
 
+    def add_bigoh(self, prec):
+        r"""
+        Return the power series of precision at most ``prec`` obtained by
+        adding `O(q^\text{prec})` to `f`, where `q` is the variable(s).
```
Comment (mantepse) [inline src/sage/rings/lazy_series.py]: Extraordinarily minor: perhaps:
```
adding `O(q)^\text{prec}` to `f`, where `q` is the (tuple of) variable(s).
```
is more correct.  (Note that sage seems to prefer `O(q,t)^3`.)

**Review by mantepse — APPROVED** [review]
feel free to ignore the comment as you wish and set it to positive review after your decision.

File: src/sage/rings/lazy_series.py (hunk 1 above)
Comment (tscrim) [reply src/sage/rings/lazy_series.py]: Done (I just copied it from the other methods, but I agree that this is better.)

**Review by mantepse — APPROVED** [review]
Thank you!

**Comment by tscrim** [comment]
Thank you.

**Pull request merged**